### PR TITLE
feat: Plugin vs. Spec

### PR DESCRIPTION
**Pull request by disq**
For initial review. ~The `Warnings` part I don't like but it can be improved with suggestions.~

### CODE REVIEW

**Comment by hermanschaaf** [comment]
Seems like a pretty decent backwards-compatible solution 👍

File: specs/spec.go [hunk 1]
```diff
@@ -11,8 +11,10 @@ import (
 type Kind int
 
 type Spec struct {
-	Kind Kind        `json:"kind"`
-	Spec interface{} `json:"spec"`
+	Kind   Kind        `json:"kind"`
+	Plugin interface{} `json:"plugin"`
+
+	Warnings []string `json:"-"`
```
Comment (hermanschaaf) [inline specs/spec.go:17]: Another option, if we don't like `Warnings` on this struct (which is a bit weird, I agree), but want to maintain backwards-compatibility in the SDK, is we can return a custom Error object with a `Warnings` attribute from `SpecUnmarshalYamlStrict`

File: specs/spec.go [hunk 2]
```diff
@@ -65,29 +70,38 @@ func (s *Spec) UnmarshalJSON(data []byte) error {
...
 	}
-	b, err := json.Marshal(t.Spec)
+
+	var (
+		b   []byte
+		err error
+	)
+	if t.Spec != nil {
+		b, err = json.Marshal(t.Spec)
+		s.Warnings = append(s.Warnings, `"spec" keyword is deprecated at this level and will be removed in a future version. Use "plugin" instead`)
```
[the remark this answers is not in the document]
Comment (disq) [reply specs/spec.go]: done :)

File: specs/spec.go (hunk 1 above)
Comment (disq) [reply specs/spec.go:17]: `error` but not an error? that's also equally weird (and even unclear maybe?)

**Comment by disq** [comment]
> Seems like a pretty decent backwards-compatible solution 👍

I almost renamed the inner-level `spec` to be `plugin` as well... and cmd+z didn't work so had to revert by hand.

**Review by yevgenypats — CHANGES_REQUESTED** [review]
I don't think we need any warnings at this stage. developing warnings parsing system for keyword deprecation in json is out of scope for this PR.

**Comment by disq** [comment]
> I don't think we need any warnings at this stage. developing warnings parsing system for keyword deprecation in json is out of scope for this PR.

removed

**Pull request closed without merging**